mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -780,12 +781,23 @@ public void updateAnnotations(MailboxPath mailboxPath, MailboxSession session, L
         for (MailboxAnnotation annotation : mailboxAnnotations) {
             if (annotation.isNil()) {
                 annotationMapper.deleteAnnotation(mailboxId, annotation.getKey());
-            } else {
+            } else if (isValid(mailboxId, annotation, annotationMapper)) {
                 annotationMapper.insertAnnotation(mailboxId, annotation);
             }
         }
     }
 
+    private boolean isValid(MailboxId mailboxId, MailboxAnnotation annotation, AnnotationMapper annotationMapper) throws AnnotationException {
+        if (annotation.size() > MailboxConstants.LIMIT_ANNOTATION_DATA) {
```

Comment:
Does annotation.size() verifies size of the name + value or just value ?

We should check RFC...

If it is not the case I can still abuse the implementation, and store arbitrary amount of data using annotations names.